cql/src/main/java/org/apache/cassandra/stargate/metrics/ClientMetrics.java

```diff
@@ -88,64 +102,114 @@ public void unpauseConnection() {
+    // netty gauges
+    meterRegistry.gauge(
+        metric("NettyDirectMemory"), Tags.empty(), this, ClientMetrics::nettyDirectMemory);
+    meterRegistry.gauge(
+        metric("NettyHeapMemory"), Tags.empty(), this, ClientMetrics::nettyHeapMemory);
 
-    authSuccess = registerMeter("AuthSuccess");
-    authFailure = registerMeter("AuthFailure");
-    authError = registerMeter("AuthError");
+    // connected native clients with the client info tags must be a multi gauge
+    connectedNativeClients =
+        MultiGauge.builder(metric("connectedNativeClients")).register(meterRegistry);
+    connectedNativeClientsByUser =
+        MultiGauge.builder(metric("connectedNativeClientsByUser")).register(meterRegistry);
 
-    pausedConnections = new AtomicInteger();
-    registerGauge("PausedConnections", pausedConnections::get);
-    requestDiscarded = registerMeter("RequestDiscarded");
-    requestsProcessed = registerMeter("RequestsProcessed");
+    pausedConnections =
+        meterRegistry.gauge(metric("PausedConnections"), Tags.empty(), new AtomicInteger(0));
 
-    totalBytesRead = registerCounter("TotalBytesRead");
-    totalBytesWritten = registerCounter("TotalBytesWritten");
+    totalBytesRead = meterRegistry.counter(metric("TotalBytesRead"));
+    totalBytesWritten = meterRegistry.counter(metric("TotalBytesWritten"));
 
-    bytesReceivedPerFrame = registerHistogram("BytesReceivedPerFrame");
-    bytesTransmittedPerFrame = registerHistogram("BytesTransmittedPerFrame");
+    bytesReceivedPerFrame = meterRegistry.summary(metric("BytesReceivedPerFrame"));
+    bytesTransmittedPerFrame = meterRegistry.summary(metric("BytesTransmittedPerFrame"));
 
     initialized = true;
+
+    // if we have the update rate, init the executor service and submit the update task
+    if (updateRateMillis > 0) {
+      executorService =
+          Executors.newSingleThreadScheduledExecutor(
+              new BasicThreadFactory.Builder()
+                  .daemon(true)
+                  .priority(1)
+                  .namingPattern("cql-metrics-updater")
+                  .build());
+      executorService.scheduleAtFixedRate(
+          () -> {
+            try {
+              updateConnectedClients();
+              updateConnectedClientsByUser();
+            } catch (Exception e) {
+              logger.warn("Error updating the connected client metrics.");
+            }
+          },
+          0,
+          updateRateMillis,
+          TimeUnit.MILLISECONDS);
+    }
+  }
+
+  /** Shuts down the #executor service and marks instance as not initialized. */
+  public synchronized void shutdown() {
+    if (!initialized) return;
+
+    if (null != executorService && !executorService.isShutdown()) {
+      executorService.shutdown();
+      try {
+        if (!executorService.awaitTermination(1, TimeUnit.SECONDS)) {
+          executorService.shutdownNow();
+        }
+      } catch (InterruptedException e) {
```

Comment:
nit: I'd log the `InterruptedException`... just for good measure and debuggability.

Reply:
Done.